src/components/AppBar.jsx

```diff
@@ -0,0 +1,173 @@
+import React, { useState, useEffect } from "react";
+import {
+  AppBar,
+  Toolbar,
+  Typography,
+  IconButton,
+  Button,
+  Avatar,
+  Menu,
+  MenuItem,
+  TextField,
+  Grid,
+} from "@mui/material";
+import SearchIcon from "@mui/icons-material/Search";
+import CancelIcon from "@mui/icons-material/Cancel";
+import { useParams, useNavigate } from "react-router-dom";
+import users from "../users";
+
+function AppBarComponent() {
+  const [isSearchOpen, setSearchOpen] = useState(false);
+  const [isDropdownOpen, setDropdownOpen] = useState(false);
+  const [avatarAnchorEl, setAvatarAnchorEl] = useState(null);
+  const [searchInput, setSearchInput] = useState("");
+  const navigate = useNavigate();
+
+  const handleSearchClick = () => {
+    setSearchOpen(!isSearchOpen);
+  };
+
+  const handleDropdownClick = (event) => {
+    setAvatarAnchorEl(event.currentTarget);
+    setDropdownOpen(true);
+  };
+
+  const closeDropdown = (event) => {
+    setDropdownOpen(false);
+    if (event.currentTarget.innerText === "Logout") {
+      navigate("/");
+    }
+  };
+  const { id } = useParams();
+
+  const user = users.find((user) => user.id.toString() === id);
+
+  const handleLinkClick = (route) => {
+    if (route === "search" && searchInput.trim() !== "") {
+      navigate(`/streamingPage/${id}/search/${searchInput}`);
+    } else if (route !== "search") {
+      navigate(`/streamingPage/${id}/${route}`);
+    }
+  };
+
+  return (
+    <AppBar
```

Comment:
If you want to style the element. do not use sx but use styled.
Ill explain you everything when youll reach this comment